### PR TITLE
Add conditions to keystoneservice and initialize them

### PR DESCRIPTION
**Pull request by stuggi**
* Adds conditions to keystoneservice
* initializes conditions used by keystoneservice and keystoneapi at beginning to unknown status
* updates kesytoneapi deployment conditon to be true when `ReadyCount > 0`

### CODE REVIEW

File: api/v1beta1/conditions.go [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+	KeystoneServiceOSUserReadyInitMessage = "Keystone Service user registration not started"
+
+	// KeystoneServiceOSUserReadyMessage
+	KeystoneServiceOSUserReadyMessage = "Keystone Service user %s ready"
+
+	// KeystoneServiceOSUserReadyWaitingMessage
+	KeystoneServiceOSUserReadyWaitingMessage = "Keystone Service user not yet ready"
+
+	// KeystoneServiceOSUserReadyErrorMessage
+	KeystoneServiceOSUserReadyErrorMessage = "Keystone Endpoints error occured %s"
```
Comment (abays) [inline api/v1beta1/conditions.go:122]: The value here looks like it should say something different.  Refer to service user instead of endpoints?

Comment (stuggi) [reply api/v1beta1/conditions.go:122]: right, good catch. will update

Comment (stuggi) [reply api/v1beta1/conditions.go]: done

**Review by abays — APPROVED** [review]
/lgtm

**Pull request merged**